modPrecompiled/src/org/aion/precompiled/ContractFactory.java

```diff
@@ -56,25 +78,33 @@ public ContractFactory() {}
      */
     @Override
     public IPrecompiledContract getPrecompiledContract(
-            ExecutionContext context,
-            IRepositoryCache<AccountState, IDataWord, IBlockStoreBase<?, ?>> track) {
+        ExecutionContext context,
+        IRepositoryCache<AccountState, IDataWord, IBlockStoreBase<?, ?>> track) {
 
         switch (context.address().toString()) {
-            case TOTAL_CURRENCY:
-                // return new TotalCurrencyContract(track, context.sender(), Address.wrap(OWNER));
+            case ADDR_TOTAL_CURRENCY:
+                //return new TotalCurrencyContract(track, context.sender(), Address.wrap(ADDR_OWNER));
                 return null;
-            case TOKEN_BRIDGE:
+            case ADDR_TOKEN_BRIDGE:
                 TokenBridgeContract contract =
-                        new TokenBridgeContract(
-                                context,
-                                track,
-                                Address.wrap(TOKEN_BRIDGE_INITIAL_OWNER),
-                                Address.wrap(TOKEN_BRIDGE));
+                    new TokenBridgeContract(
+                        context,
+                        track,
+                        Address.wrap(ADDR_TOKEN_BRIDGE_INITIAL_OWNER),
+                        Address.wrap(ADDR_TOKEN_BRIDGE));
 
-                if (!context.origin().equals(Address.wrap(TOKEN_BRIDGE_INITIAL_OWNER))
-                        && !contract.isInitialized()) return null;
+                if (!context.origin().equals(Address.wrap(ADDR_TOKEN_BRIDGE_INITIAL_OWNER))
+                    && !contract.isInitialized()) {
+                    return null;
+                }
 
                 return contract;
+            case ADDR_ED_VERIFY:
+                return PC_ED_VERIFY;
+            case ADDR_BLAKE2B_HASH:
+                return PC_BLAKE2B_HASH;
+            case ADDR_TX_HASH:
+                return new TXHashContract(context);
```

Comment:
what's the reasoning behind handling the TX contract differently?

Reply:
cause the TxHash need the context setup.